hash/hash_comparator.go

```diff
@@ -2,54 +2,134 @@ package hash
 
 import (
 	"context"
+	"crypto/aes"
+	"crypto/cipher"
 	"crypto/subtle"
 	"encoding/base64"
+	"encoding/hex"
 	"fmt"
-	"regexp"
 	"strings"
 
 	"github.com/pkg/errors"
 	"golang.org/x/crypto/argon2"
 	"golang.org/x/crypto/bcrypt"
+	"golang.org/x/crypto/sha3"
 
 	"github.com/ory/kratos/driver/config"
 )
 
 var ErrUnknownHashAlgorithm = errors.New("unknown hash algorithm")
+var ErrUnknownHashFormat = errors.New("unknown hash format")
+var ErrNonceSizeIncorrect = errors.New("nonce size is incorrect")
```

Comment:
These errors are actually already annotated with a stack, but it is basically the `init` functions stack. IMO, we should change them to
```suggestion
var ErrUnknownHashAlgorithm = fmt.Errorf("unknown hash algorithm")
var ErrUnknownHashFormat = mt.Errorf("unknown hash format")
var ErrNonceSizeIncorrect = mt.Errorf("nonce size is incorrect")
```